### PR TITLE
integration: Add the re-usable "retry" helper.

### PR DESCRIPTION
**Pull request by aluzzardi**
Retry was kind of extracted from wait_until_reachable. The goal is to be
able to wait for a command to succeed, or fail if it doesn't succeed
after a defined number of attempts.

Indirectly improve the error logging: the output of the retry command
won't show up unless it constantly failed.

Signed-off-by: Andrea Luzzardi <aluzzardi@gmail.com>

### CODE REVIEW

**Comment by aluzzardi** [comment]
/cc @jimmyxian 

Can you verify that the integration tests all work on your side with this PR?

**Comment by vieux** [comment]
LGTM

**Comment by abronan** [comment]
LGTM as well, waiting to hear back from @jimmyxian ;)

**Comment by jimmyxian** [comment]
@aluzzardi @abronan @vieux 
Great! I will try to use it..

File: test/integration/helpers.bash [hunk 1]
```diff
@@ -36,15 +36,29 @@ function swarm() {
...
+	local delay=$1
+	shift
+	local i
+
+	for ((i=0; i < attempts; i++)); do
+		run "$@"
+		if [[ "$status" -eq 0 ]] ; then
+			return 0
+		fi
+		sleep $delay
```
Comment (jimmyxian) [inline test/integration/helpers.bash]: for ((i=0; i < attempts; i++)); do
                sleep $delay
		run "$@"
		if [[ "$status" -eq 0 ]] ; then
			return 0
		fi
		
maybe we should change the 'sleep' position, put sleep function first. The test will be ok!

**Pull request merged**